bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/contentassist/AsyncCompletionProposalPopup.java

```diff
@@ -389,11 +419,46 @@ protected List<CompletableFuture<List<ICompletionProposal>>> buildCompletionFutu
 					return Collections.emptyList();
 				}
 				return proposals;
-			}));
+			}, EXECUTOR));
 		}
 		return futures;
 	}
 
+	/**
+	 * Submit a task in such a way that it actually reacts to cancellation (i.e. calls to
+	 * {@code future.cancel(true)})
+	 *
+	 * @param executor Do not use the common pool here since that one does not cancel (interrupts)
+	 *            worker threads
+	 * @return an interruptible future.
+	 */
+	private static <T> CompletableFuture<T> submitInterruptible(
+			Callable<T> task, ExecutorService executor) {
+
+		CompletableFuture<T> cf= new CompletableFuture<>();
+		FutureTask<T> ft= new FutureTask<>(task);
+
+		executor.submit(() -> {
+			try {
+				ft.run(); // executes in this thread
+				cf.complete(ft.get());
```

Comment:
Executor submit already returns a `Future` why do we need a separate one here?

One would expect to simply submit a runnable as below that simply calls the callable directly.


```suggestion
		Future<T> ft = executor.submit(() -> {
			try {
				cf.complete(task.call());
```

Reply:
Done